### PR TITLE
tweak 404 handling

### PR DESCRIPTION
**Pull request by dobbymoodge**

### CODE REVIEW

File: test_pull_requests [hunk 1]
```diff
@@ -322,13 +322,11 @@ module Hub
...
-          res = delete "#{GITHUB_API_BASE_URL}/repos/#{Properties['github_user']}/#{repo}/issues/comments/%s" % [comment_id]
-        rescue Net::HTTPNotFound => e
-          # If we get 404 on a retry, that means there's no comment to delete
-          break if i > 1
-        end
+        res = delete "#{GITHUB_API_BASE_URL}/repos/#{Properties['github_user']}/#{repo}/issues/comments/%s" % [comment_id]
+        # If we get 404 on a retry, that means there's no comment to delete
         break if res.success?
+        # convert to string because it looks like res.code could be int or string depending on ruby
+        break if ("#{res.code}" == "404" && i > 1)
```
Comment (danmcp) [inline test_pull_requests]: What type is res.code?

Comment (dobbymoodge) [inline test_pull_requests]: in my repl it gives me a string. In the other place it's called from in test_pull_requests, it's compared against an int (until this PR)

Comment (danmcp) [inline test_pull_requests]: Can you find the doc that shows it changed from one to another?

At a min use res.code.to_i  == 404

Comment (dobbymoodge) [inline test_pull_requests]: huh, according to the docs [here](http://ruby-doc.org/stdlib-1.9.3/libdoc/net/http/rdoc/Net/HTTPResponse.html) and [here](http://ruby-doc.org/stdlib-2.1.1/libdoc/net/http/rdoc/Net/HTTPResponse.html) it should be a string. I'll fix the code accordingly.

**Comment by dobbymoodge** [comment]
@danmcp should be correct now

File: test_pull_requests [hunk 2]
```diff
@@ -322,13 +322,10 @@ module Hub
...
       (1..num_tries).each do |i|
-        begin
-          res = delete "#{GITHUB_API_BASE_URL}/repos/#{Properties['github_user']}/#{repo}/issues/comments/%s" % [comment_id]
-        rescue Net::HTTPNotFound => e
-          # If we get 404 on a retry, that means there's no comment to delete
-          break if i > 1
-        end
+        res = delete "#{GITHUB_API_BASE_URL}/repos/#{Properties['github_user']}/#{repo}/issues/comments/%s" % [comment_id]
+        # If we get 404 on a retry, that means there's no comment to delete
         break if res.success?
```
Comment (danmcp) [inline test_pull_requests]: Put the two breaks into 1 line with an ||

Comment (dobbymoodge) [inline test_pull_requests]: eh, I like this better but it's not worth debating

Comment (dobbymoodge) [inline test_pull_requests]: done

File: test_pull_requests [hunk 3]
```diff
@@ -322,13 +322,9 @@ module Hub
...
-        begin
-          res = delete "#{GITHUB_API_BASE_URL}/repos/#{Properties['github_user']}/#{repo}/issues/comments/%s" % [comment_id]
-        rescue Net::HTTPNotFound => e
-          # If we get 404 on a retry, that means there's no comment to delete
-          break if i > 1
-        end
-        break if res.success?
+        res = delete "#{GITHUB_API_BASE_URL}/repos/#{Properties['github_user']}/#{repo}/issues/comments/%s" % [comment_id]
+        # If we get 404 on a retry, that means there's no comment to delete
+        break if res.success? || (res.code == "404" && i > 1)
```
Comment (danmcp) [inline test_pull_requests]: nit, but single quotes would be preferred

**Comment by dobbymoodge** [comment]
@danmcp nit picked

**Pull request merged**